src/app/shared/components/password-check-tooltip/password-check-tooltip.component.spec.ts

```
@@ -0,0 +1,153 @@
+    it('should fail when password length is less than 12 characters', () => {
+      const emitSpy = spyOn(component.isPasswordValid, 'emit');
+      component.password = 'Short1!';
+      fixture.detectChanges();
+      component.validatePassword();
+      expect(component.passwordChecks.lengthValid).toBeFalse();
+      expect(emitSpy).toHaveBeenCalledWith(false);
+    });
+
+    it('should fail when password length is more than 32 characters', () => {
+      const emitSpy = spyOn(component.isPasswordValid, 'emit');
+      component.password = 'A'.repeat(33) + '1!';
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)